src/acquisition/flusurv/flusurv_update.py

```diff
@@ -113,38 +112,185 @@ def update(issue, location_name, test_mode=False):
+      %(rate_age_4)s,
+      %(rate_age_5)s,
+      %(rate_age_6)s,
+      %(rate_age_7)s,
+
+      %(rate_age_18t29)s,
+      %(rate_age_30t39)s,
+      %(rate_age_40t49)s,
+      %(rate_age_5t11)s,
+      %(rate_age_12t17)s,
+      %(rate_age_<18)s,
+      %(rate_age_>=18)s,
+
+      %(rate_race_white)s,
+      %(rate_race_black)s,
+      %(rate_race_hispaniclatino)s,
+      %(rate_race_asianpacificislander)s,
+      %(rate_race_americanindianalaskanative)s,
+
+      %(rate_sex_male)s,
+      %(rate_sex_female)s
     )
     ON DUPLICATE KEY UPDATE
-      `release_date` = least(`release_date`, %s),
-      `rate_age_0` = coalesce(%s, `rate_age_0`),
-      `rate_age_1` = coalesce(%s, `rate_age_1`),
-      `rate_age_2` = coalesce(%s, `rate_age_2`),
-      `rate_age_3` = coalesce(%s, `rate_age_3`),
-      `rate_age_4` = coalesce(%s, `rate_age_4`),
-      `rate_overall` = coalesce(%s, `rate_overall`),
-      `rate_age_5` = coalesce(%s, `rate_age_5`),
-      `rate_age_6` = coalesce(%s, `rate_age_6`),
-      `rate_age_7` = coalesce(%s, `rate_age_7`)
+      `release_date` = least(`release_date`, %(release_date)s),
+      `rate_overall` = coalesce(%(rate_overall)s, `rate_overall`),
+
+      `rate_age_0` = coalesce(%(rate_age_0)s, `rate_age_0`),
+      `rate_age_1` = coalesce(%(rate_age_1)s, `rate_age_1`),
+      `rate_age_2` = coalesce(%(rate_age_2)s, `rate_age_2`),
+      `rate_age_3` = coalesce(%(rate_age_3)s, `rate_age_3`),
+      `rate_age_4` = coalesce(%(rate_age_4)s, `rate_age_4`),
+      `rate_age_5` = coalesce(%(rate_age_5)s, `rate_age_5`),
+      `rate_age_6` = coalesce(%(rate_age_6)s, `rate_age_6`),
+      `rate_age_7` = coalesce(%(rate_age_7)s, `rate_age_7`),
+
+      `rate_age_18t29` = coalesce(%(rate_age_18t29)s, `rate_age_18t29`),
+      `rate_age_30t39` = coalesce(%(rate_age_30t39)s, `rate_age_30t39`),
+      `rate_age_40t49` = coalesce(%(rate_age_40t49)s, `rate_age_40t49`),
+      `rate_age_5t11` = coalesce(%(rate_age_5t11)s, `rate_age_5t11`),
+      `rate_age_12t17` = coalesce(%(rate_age_12t17)s, `rate_age_12t17`),
+      `rate_age_lt18` = coalesce(%(rate_age_<18)s, `rate_age_lt18`),
+      `rate_age_gte18` = coalesce(%(rate_age_>=18)s, `rate_age_gte18`),
+
+      `rate_race_white` = coalesce(%(rate_race_white)s, `rate_race_white`),
+      `rate_race_black` = coalesce(%(rate_race_black)s, `rate_race_black`),
+      `rate_race_hisp` = coalesce(%(rate_race_hispaniclatino)s, `rate_race_hisp`),
+      `rate_race_asian` = coalesce(%(rate_race_asianpacificislander)s, `rate_race_asian`),
+      `rate_race_natamer` = coalesce(%(rate_race_americanindianalaskanative)s, `rate_race_natamer`),
+
+      `rate_sex_male` = coalesce(%(rate_sex_male)s, `rate_sex_male`),
+      `rate_sex_female` = coalesce(%(rate_sex_female)s, `rate_sex_female`)
     """
 
     # insert/update each row of data (one per epiweek)
     for epiweek in epiweeks:
         lag = delta_epiweeks(epiweek, issue)
-        if lag > 52:
+        if lag > max_age_to_consider_weeks:
```

Comment:
does the `seasonids` creation/filtering negate the need for this check?

Reply:
No, they're not exactly the same. Each seasonid corresponds to a flu season, which spans several months. So some data from a season could fall within the 52-week cutoff we're using and some could be outside that. The 52-week filter is required; the seasonid filter is not.

The seasonid filtering happens before we make any API calls. Given that the data spans a long time period (back to 2009?), I didn't want to fetch its whole history. Realistically, since this is weekly data, fetching only recent seasons probably doesn't make a big difference on runtime or memory usage. Just a preference, but I'm open to feedback on this. It does add complexity.